tests/Unit/Domain/Creators/Test_CylindricalBinaryCompactObject.cpp

```diff
@@ -283,9 +284,63 @@ void test_connectivity_once(const bool with_sphere_e,
     CHECK(binary_compact_object.block_names() == block_names);
     CHECK(binary_compact_object.block_groups() == block_groups);
 
-    TestHelpers::domain::creators::test_domain_creator(
+    const auto domain = TestHelpers::domain::creators::test_domain_creator(
         binary_compact_object, with_boundary_conditions);
 
+    std::unordered_map<std::string, ExcisionSphere<3>>
+        expected_excision_spheres{};
+    std::unordered_map<size_t, Direction<3>> abutting_directions_A;
+    if(include_inner_sphere_A) {
+      for (size_t i = 46; i < 56; ++i) {
+        abutting_directions_A.emplace(i, Direction<3>::lower_zeta());
+      }
+      for (size_t i = 56; i < 60; ++i) {
+        abutting_directions_A.emplace(i, Direction<3>::lower_xi());
+      }
+    } else {
+      for (size_t i = 9; i < 14; ++i) {
+        abutting_directions_A.emplace(i, Direction<3>::lower_zeta());
+      }
+      for (size_t i = 14; i < 18; ++i) {
+        abutting_directions_A.emplace(i, Direction<3>::lower_xi());
+      }
+      for (size_t i = 27; i < 32; ++i) {
+        abutting_directions_A.emplace(i, Direction<3>::lower_zeta());
+      }
+    }
+    std::unordered_map<size_t, Direction<3>> abutting_directions_B;
+    if(include_inner_sphere_B) {
+      const size_t first_inner_sphere_block = include_inner_sphere_A ? 60 : 46;
+      for (size_t i = first_inner_sphere_block;
+           i < first_inner_sphere_block + 10; ++i) {
```

Comment:
This test just reimplements the code that we're testing, doesn't it? That's not super useful and hard to maintain. Can we add some consistency checks to `test_domain_creator` instead, e.g. that the block has an external boundary in that direction etc?

Reply:
I realize that this test (like many other tests) are re-implementations of the same code in a different way, but I'm not sure what you want me to do. This is basically the same excision_spheres test as in Test_BinaryCompactObject (but the case here is a bit more complicated).  `test_domain_creator` is general, and the property of which blocks have which excision boundaries in which directions are domain-dependent, e.g. they are different for BinaryCompactObject and CylindricalBinaryCompactObject, and they are different depending on the options.

Reply:
My suggestion is to replace this reimplementation with a consistency check in `test_domain_creator` that all `abutting_directions()` in all `DomainCreator::excision_spheres()` are also `Block::external_boundaries()`. That should be enough. Maybe there's a way to make sure that the excision sphere is fully covered with blocks, but I can't think of a nice way to do that right now.

Reply:
And you're right about the "like many other tests": reimplementations are done all over the place in the domain creator tests, which make them very hard to maintain. I opened an issue for that here: #4408